packages/client/src/execution/vmexecution.ts

```diff
@@ -151,38 +151,46 @@ export class VMExecution extends Execution {
    * the entire procedure.
    * @param receipts If we built this block, pass the receipts to not need to run the block again
    */
-  async runWithoutSetHead(opts: RunBlockOpts, receipts?: TxReceipt[]): Promise<void> {
-    return this.runWithLock<void>(async () => {
-      const { block, root } = opts
-
-      if (receipts === undefined) {
-        // Check if we need to pass flag to clear statemanager cache or not
-        const prevVMStateRoot = await this.vm.stateManager.getStateRoot()
-        // If root is not provided its mean to be run on the same set state
-        const parentState = root ?? prevVMStateRoot
-        const clearCache = !equalsBytes(prevVMStateRoot, parentState)
+  async runWithoutSetHead(opts: RunBlockOpts, receipts?: TxReceipt[]): Promise<boolean> {
+    if (this.running || !this.started || this.config.shutdown) return false
+    this.running = true
 
-        const result = await this.vm.runBlock({ clearCache, ...opts })
```

Comment:
most of diff here is by introduction of `try{...}finally{this.running=false}` pattern